src/arrayMethodJoin.js

```diff
@@ -4,8 +4,31 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    if (Array.isArray(this) && this.length > 1) {
+      let str;
+
+      if (this[0] === null || this[0] === undefined) {
+        str = '';
+      } else {
+        str = `${this[0]}`;
+      }
+
+      let arrValue;
+
+      for (let i = 1; i < this.length; i++) {
+        if (this[i] === null || this[i] === undefined) {
+          arrValue = '';
+        } else {
+          arrValue = `${this[i]}`;
```

Comment:
When you add element to a string, it becomes string too. No need to use interpolation here.

Reply:
fixed